scipy/optimize/_linprog_highs.py

```diff
@@ -424,7 +424,7 @@ def _linprog_highs(lp, solver, time_limit=None, presolve=True,
            }
 
     if np.any(x) and integrality is not None:
-        res.update({
+        sol.update({
```

Comment:
This used to be updating the `res` object which never gets returned, so these values would never show up in the result even if the condition in the if statement was true. Fixed by changing to `sol.update`

Reply:
Huh, good call. It would be good to add a test that `mip_node_count` and `mip_dual_bound` are also present in the result object, if you're willing.

Reply:
Done!